packages/front-end/components/Segments/SegmentForm.tsx

```diff
@@ -70,7 +71,9 @@ const SegmentForm: FC<{
         placeholder="Choose one..."
         options={filteredDatasources.map((d) => ({
           value: d.id,
-          label: d.name,
+          label:
+            d.name +
+            (d.description ? ` — ${truncateText(d.description, 80)}` : ""),
```

Comment:
Does this work for all screen sizes? e.g. smaller ones? The CSS property `text-overflow: ellipsis` is more dynamic and would work with a variety of widths. https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow